### DISCUSSION TITLE
PinMapping.json download?

### DISCUSSION THREAD

**Discussion by 3joern** [hoylabs/OpenDTU-OnBattery#1243] [Q&A] [answered]
Hallo Zusammen,
nachdem ich vor 3 Wochen versucht habe, mein 4mb ESP32-D0WD-V3 mit der neusten Firmware zu beglücken, [das aber richtig schief ging](https://github.com/helgeerbe/OpenDTU-OnBattery/discussions/1166#discussioncomment-10373124), hab ich mich nochmal versucht einzulesen und bin auf das PinMapping gestoßen. 
Kann das Problem daran gelegen haben, dass zwar die DTU gestartet ist, aber keine Verbindung zum Wechselrichter aufgebaut werden konnte? (Siehe Screenshot im Link oben).
Bisher bin ich immer davon ausgegangen ich lade die config.json runter, aktualisiere das Ding per USB, lade die config wieder hoch und alles ist wie gehabt. Ich lese nun die Doku und das wiki rauf und runter, aber ich komm nicht drauf, was ich falsch gemacht habe bis auf das es da noch ein PinMapping gibt. Hab aber in meiner Oberfläche keine Möglichkeit das pin mapping runterzuladen. 
Sieht aktuell so aus:

![Bildschirmfoto 2024-09-09 um 15 35 12](https://github.com/user-attachments/assets/6d76c878-7a2a-41cf-9fde-3ebd8652aa45)

![Bildschirmfoto 2024-09-09 um 15 35 23](https://github.com/user-attachments/assets/c74ffb12-a667-4e56-8ccd-bb0cb1000edb)

![Bildschirmfoto 2024-09-09 um 15 35 36](https://github.com/user-attachments/assets/681a14b6-4b1e-4c95-8993-9ac53881167e)

![Bildschirmfoto 2024-09-09 um 15 35 40](https://github.com/user-attachments/assets/9ef23d89-88f9-4bbf-b250-b12d3b9cc100)

Ich las was von einem generischen pin Mapping, da hätte ich aber schiss das einfach mal so drüber zu bügeln und das DTU komplett zu zerschiessen. Könntet Ihr mir bitte kurz einen Hinweis geben, woran es gelegen haben könnte oder wonach ich schauen soll? Danke :)

**Comment by spcqike** [comment]
Das Mapping kannst du in der Verwaltung runterladen 

<img width="375" alt="image" src="https://github.com/user-attachments/assets/7a878a62-98db-4ba4-a026-2984fbd3978a">


Eins deiner Probleme wird auch sein, dass unter anderen Pin 18 doppelt vergeben ist. Für den nrf und für eth. 

Erstell die ein sauberes Mapping, welcher deiner Hardware entspricht, und lade es hoch.

**Reply by 3joern** [reply]
Ich wäre auch davon ausgegangen, dass ich es da runter laden könnte, geht aber nicht, habe nur die config.json zur Auswahl.
Hochladen habe ich beide Optionen.
![Bildschirmfoto 2024-09-09 um 17 04 51](https://github.com/user-attachments/assets/3781e59d-36bd-4211-b362-5af3c3d768af)


Zu der Pinvergabe, ich habe das DTU fertig mit damals der normalen "open DTU" gekauft. Hab kein Schimmer, auch sind Display Sachen aktiv, obwohl es kein Display gibt.

**Reply by spcqike** [reply]
Kann es sein dass dann in der Firmware feste Pins definiert waren? 

Wenn noch kein gültiges pinmapping hochgeladen ist, wirst du auch keines runter laden können. 

Erstell die eins und lade es hoch. Würd ich zumindest einfach versuchen.

**Comment by 3joern** [comment, marked as answer]
So habs hinbekommen mit der Hilfe von Jürgen vom dtushop.de, bei dem ich vor nem Jahr das Gerät gekauft habe. Gott bin ich dankbar für so Verkäufer, die einen dann anrufen und sagen was man machen soll. 
Hab alles wie das letzte mal gemacht und wie das letzte mal konnte die DTU wieder keine Verbindung zum WR aufbauen, diesmal hatte ich aber das pinmapping.json und habs auch hochgeladen, **die Info, dass ich das aber auch noch bei Hardware unter Ausgewähltes Profil auswählen muss**, habe ich aber irgendwie nicht mitbekommen, nachdem ich das Profil mit dem richtigen Pin Mapping nun ausgewählt habe funktioniert alles so wie soll.

Längerfristig werd ich mir aber auf jedenfall die DTU mit 16mb von Jürgen holen (das ich OTA nutzen kann) und die 4mb DTU für meinen BluePalm Speicher ummodeln, dass ich den Speicher ohne China Cloud steuern kann.

**Reply by Sorbat** [reply]
wenn du dir sowieso eine neue dtu holen willst für OTA, warum dann nicht direkt das Fusion?

**Reply by 3joern** [reply]
warum, was kann das?

**Reply by Sorbat** [reply]
die frage sollte doch eher sein: was kann es nicht :)
<img width="785" alt="overview" src="https://github.com/user-attachments/assets/3822ce36-59a5-483d-8ed5-3492d4052b4c">
hier die Doku: https://github.com/markusdd/OpenDTUFusionDocs
und hier zu erwerben: 
https://shop.allianceapps.io/collections/hauptprodukte
https://3printd-solution.eu/DTU-Produkte

**Reply by 3joern** [reply]
Ich bin eher Endanwender und nicht ganz so der IT Pro, aktuell kann ich noch kein Vorteil sehen, kann ich da zB zwei Partitionen drauf laufen lassen? einmal für OpenDTUoB und einmal um den B2500 per MQTT zu steuern? was kann das etwas teurere fusion Board mehr oder besser als das vorkonfigurierte OpenDTU mit 16mb? Ich hab nur was gefunden, dass es mehr Wechselrichterfirmen unterstütz, richtig?

**Reply by Sorbat** [reply]
mhh weiß nicht so recht, ob das teurer ist. aber wenn du das sagst, kann das durchaus sein, so genau habe ich das nicht verglichen. 

Ich kann bei den DTUs aus deinem DTUShop nicht erkennen, um welches Innenleben es sich handelt, aber oftmals gibt es hier Steckverbindungen über Jumperkabel, welche nicht für einen dauerhaft zuverlässigen Betrieb geeignet sind. 
Das Fusion wurde von Leuten aus der Community entwickelt, welche auch auf dem Discord vertreten sind und es ist zweifellos die beste Lösung, gerade auch für technisch nicht so versierte Nutzer, nicht zuletzt auch, weil hier seitens der Community ein besserer support geboten werden kann. 

Die Erfahrung, auch im Discord, zeigt, dass man durchaus auch Stunden oder Tage damit verbringen kann "irgendeine" DTU zum laufen zu bringen, oder Probleme zu beheben. Oder aber man entscheidet sich für das Fusion und hat viele dieser Probleme gar nicht erst. 
-> Das Fusion gab es nie mit weniger als 8MB Speicher. Entsprechend braucht sich mit dem Fusion auch niemand große Gedanken über die 4MB Abkündigung zu machen. Aber das ist wirklich nur ein sehr kleiner von sehr sehr vielen Punkten.

Das Fusion ist halt einfach das Maß der Dinge.

**Comment by chwarg** [comment]
Ich bin hier am verzweifeln - jedes Mal wenn ich die Pin config hoch lade, kann ich danach nicht mehr ins Hardware Menü. Das Menü läd in einer Endlosschleife aber es passiert nichts. Die einzige Möglichkeit das zu beheben ist den ESP neu zu flashen. Warum gibt es nicht die Möglichkeit im Interface ein neues Profil zu erstellen. Das ist einfach unnötig kompliziert.

**Reply by Harisjems** [reply]
In diesem Fall ist in der Regel die pin.mapping nicht korrekt.
Es stimmt aber, dass die Handhabung von pin.mapping bei AhoyDTU einfacher ist, was die Konfiguration angeht.
